msalautomationapp/src/main/java/com/microsoft/identity/client/msal/automationapp/sdk/IAuthSdk.java

```diff
@@ -0,0 +1,51 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client.msal.automationapp.sdk;
+
+import androidx.annotation.NonNull;
+
+import com.microsoft.identity.client.ui.automation.TokenRequestTimeout;
+import com.microsoft.identity.client.ui.automation.interaction.OnInteractionRequired;
+
+/**
+ * An interface describing methods of acquire token. Implementing this interface
+ * on a test facilitates specifying the necessary parameters required to acquire
+ * token either interactively or silently.
+ */
+public interface IAuthSdk {
```

Comment:
Can we move this as a generic interface to `uiautomationutilities`? that way it can used by both MSAL and ADAL automation

Reply:
you could do something like this:

```java
public interface IAuthSdk<T extends AuthTestParams> {

    /**
     * Get the token interactively that can be used to access resources.
     *
     * @return A resultant token or an exception occurred while acquiring token interactively
     */
    AuthResult acquireTokenInteractive(@NonNull final T authTestParams,
                                       final OnInteractionRequired interactionRequiredCallback, final TokenRequestTimeout tokenRequestTimeout) throws Throwable;

    /**
     * Get the token silently that can be used to access resources.
     *
     * @return A resultant token or an exception occurred while acquiring token silently
     */
    AuthResult acquireTokenSilent(@NonNull final T authTestParams, final TokenRequestTimeout tokenRequestTimeout) throws Throwable;
}
```

and then in the MSAL SDK you could do something like this:

```java
public class MsalSdk implements IAuthSdk<MsalAuthTestParams> {
 // remaining code here won't need to change
}
```

This would allow you to have a much more generic `IAuthSdk` and move it to `uiautomationutilities`

Reply:
added the generic Auth Sdk in uiautomation utilities (common) and using here now.